### ISSUE TITLE
Request: Add more context to error traces

### ISSUE THREAD

**Issue by mcandre** [biojppm/rapidyaml#399]
rapidyaml often reports `check failed: (ch != NONE)` when attempting to traverse YAML data that omits expected keys.

In large applications, most users will be confused by this message, unable to take action to correct the mistake.

Can we please present more specific error messages. For example, indicate the name of the YAML key that the application was attempting to lookup.

**Comment by Runner-2019** [comment]
Thats what I meet either. I make a wrapper for the lower level API usages. Its not easy and costs lots of time. Maybe a more detailed error trace messages should be added by ryml.

**Comment by biojppm** [comment]
I agree that the error messages could be improved. I am revisiting this in the context of #389 and I will devise a better solution.

**Comment by biojppm** [comment]
Actually this fix missed the boat on #389.

I'm thinking of adding a second error handler for node visitation:

```c++
// existing error handler: tuned for parsing
using pfn_error = void (*) (const char* msg, size_t msg_len, Location location, void *user_data);

// to be added:
using pfn_error_visit = void (*) (const char* msg, size_t msg_len, Tree const& tree, size_t id, void *user_data);
```

But this involves a thorough look at the existing calls, so it will have to wait for ongoing work to be merge in (specifically #414 which is giant).

**Comment by DhavalsinhRaj** [comment]
hey, just wanted to ask,Are there any plan to implement this in future?

**Comment by mcandre** [comment]
ah, i see this is being worked on. cool!

https://github.com/biojppm/rapidyaml/pull/503#issuecomment-2755062767

**Comment by biojppm** [comment]
@mcandre #503 is ready to merge, and it has breaking changes; if you have any feedback, now is the time to do so. Start with PR description to make it easier.

**Comment by mcandre** [comment]
Bless.

Understood about breaking changes.

Sounds like a good improvement.

I'm archiving my C/C++ projects in favor of Rust and Go.
